vecOps.cu

```diff
@@ -0,0 +1,100 @@
+#include<vector>
+#include<iostream>
+#include <chrono> // for measuring performance
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h>
+
+auto cpuVectorAddition(std::vector<int> &A, std::vector<int> &B) {
+    auto start = std::chrono::high_resolution_clock::now();
+    for(int i = 0;i< A.size();i++) {
+        A[i] += B[i];
+    }
+    auto stop = std::chrono::high_resolution_clock::now(); 
+    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(stop - start); 
+    std::cout <<"Speed of CPU vector Addition: " << duration.count() <<" micro seconds"<<std::endl; 
+    return duration.count();
+}
+
+__global__ void add(int *a, int *b, int*c) {
+    c[blockIdx.x] = a[blockIdx.x] + b[blockIdx.x];
+}
+
+// GPU vector Addition using Pointers
+auto gpuVectorAddition(std::vector<int> &A, std::vector<int> &B) {
+    size_t n= A.size();
+
+    int* h_A = A.data();
+    int* h_B = B.data();
+
+    int *d_a, *d_b, *d_c;
+    cudaMalloc((void**)&d_a, sizeof(int)*n);
+    cudaMalloc((void**)&d_b, sizeof(int)*n);
+    cudaMalloc((void**)&d_c, sizeof(int)*n);
+
+    cudaMemcpy((void *)d_a, h_A, sizeof(int)*n, cudaMemcpyHostToDevice);
+    cudaMemcpy((void *)d_b, h_B, sizeof(int)*n, cudaMemcpyHostToDevice);
+
+    // Timing stuff, record how many seconds it takes for this operation
+    cudaEvent_t launch_begin, launch_end;
+    cudaEventCreate(&launch_begin);
+    cudaEventCreate(&launch_end);
+
+    // Warmup
+    add<<<n,1>>>(d_a, d_b, d_c);// num blocks, num_threads
```

Comment:
Maybe we can add a boolean parameter to the function which controls whether we want to run it 100 times and get the average, or just run it once? It can perhaps have a default value set to 1 run.

Reply:
SO in a GPU function, which add is (signified by the <<< >>> operators), we can't signify how many times we want the kernel to run. We would have to probably wrap this in a separate function which controls the number of times this function is run. Should I do that ?

Reply:
And before testing timing on the GPU you have to do one warmup. How do you think this needs to be structured ?

Reply:
I don't have enough context on GPU programming to answer this well.

My comment was more from a high-level perspective, that instead of running it 100 times for each function call, maybe just have a switch which controls the number of times the same operation is run (irrespective of warmup).

Reply:
Okay I'll do that man

Reply:
So I was testing the vector addition and some bugs were coming up that will take some time to fix. So I've deleted the vecOps.cu file containing the cuda stuff. I'll work on that separately in another branch. I've fixed everything else. Thanks for the great review!